src/brod_utils.erl

```diff
@@ -443,7 +460,7 @@ do_fetch_committed_offsets(Conn, GroupId, Topics) when is_pid(Conn) ->
 %% @doc Fetch a message-set. If the given MaxBytes is not enough to fetch a
 %% single message, expand it to fetch exactly one message
 -spec fetch(connection(), req_fun(), offset(), kpro:count()) ->
-               {ok, {offset(), [brod:message()]}} | {error, any()}.
+               {ok, {offset(), offset(), [brod:message()]}} | {error, any()}.
 fetch(Conn, ReqFun, Offset, MaxBytes) ->
```

Comment:
I mean something like this:
```
fetch(Conn, ReqFun, Offset, MaxBytes, AllowEmptyBatch) ->
  case do_fetch(Conn, ReqFun, Offset, MaxBytes) of
    {ok, {StableOffset, NextOffset, []}} when not AllowEmptyBatch andalso NextOffset < StableOffset ->
      fetch(Conn, ReqFun, NextOffset, MaxBytes, AllowEmptyBatch);
    Other ->
      Other
end.

fetch(Conn, ReqFun, Offset, MaxBytes) ->
  case do_fetch(Conn, ReqFun, Offset, MaxBytes) of
    {ok, {StableOffset, _NextOffset, Msgs}} ->
      {ok, {StableOffset, Msgs}}; %% for backward compatibility
    Other ->
      Other
end.
```

so the caller will not have to deal with empty batches

Reply:
To clarify, there already is brod_utils:fetch/5 that is also exported:
```
fetch(Hosts, Topic, Partition, Offset, Opts) when is_list(Hosts) ->
fetch({Hosts, ConnCfg}, Topic, Partition, Offset, Opts) ->
fetch(Client, Topic, Partition, Offset, Opts) when is_atom(Client) ->
fetch(Conn, Topic, Partition, Offset, Opts) ->
```
Under the hood, it uses make_fetch_fun, which uses fetch/4.

Is it okay to add a new fetch/5 with quite a different signature? I want to be sure, as I don't write erlang daily. Also, do you think AllowEmptyBatch should be replaced with a list of options instead of a boolean flag? I'm okay with doing it either way.

Reply:
Right, sorry for missing that.
maybe `fetch_one_batch` ?

Reply:
@zmstone I got distracted, but finally addressed your comments